GetandTake/Program.cs

```diff
@@ -1,10 +1,24 @@
+using AutoMapper;
+using GetandTake.DataAccess.Repositories;
+using GetandTake.DataAccessLayer.EF;
+using GetandTake.Services.Abstracts;
+using GetandTake.Services.AutoMapper;
+using GetandTake.Services.Concretes;
+using Microsoft.EntityFrameworkCore;
+using System;
+
 var builder = WebApplication.CreateBuilder(args);
 
+builder.Services.AddDbContext<NorthwindDbContext>(
+    options =>  options.UseSqlServer(builder.Configuration.GetConnectionString("DefaultConnection")));
 // Add services to the container.
 builder.Services.AddRazorPages();
+builder.Services.AddTransient<IProductService, ProductManager>();
+builder.Services.AddTransient<ICategoryService, CategoryManager>();
+builder.Services.AddTransient(typeof(IEntityRepository<>), typeof(EFEntityRepository<>));
+builder.Services.AddSingleton(new MapperConfiguration(x => x.AddProfile(new AutoMapperProfile())).CreateMapper());
```

Comment:
Unfotrunately, `x` is not really obvious. Let's specify full name of your delegate variable (I suggest it should be like `mapperConfig`)